<a href="https://colab.research.google.com/github/ipsilon7/Analisis-de-datos-de-IMDb/blob/main/Proceso_ETL.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Configuracion del entorno**

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

# URL del archivo en GitHub
url = 'https://raw.githubusercontent.com/ipsilon7/Analisis-de-datos-de-IMDb/refs/heads/main/Database/IMDb%20database.csv'

# Cargar el archivo CSV desde GitHub
df = pd.read_csv(url)

print(df.dtypes)
print(f"Uso de memoria: {df.memory_usage(deep=True).sum() / (1024**2):.2f} MB")

id                object
title             object
type              object
genres            object
averageRating    float64
numVotes         float64
releaseYear      float64
dtype: object
Uso de memoria: 289.19 MB


id                object
<br>
title             object
<br>
type              object<br>
genres            object<br>
averageRating    float64<br>
numVotes         float64<br>
releaseYear      float64<br>
dtype: object<br>
Uso de memoria: 289.19 MB

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

# URL del archivo en GitHub
url = 'https://raw.githubusercontent.com/ipsilon7/Analisis-de-datos-de-IMDb/refs/heads/main/Database/IMDb%20database.csv'

# Cargar el archivo CSV desde GitHub
df = pd.read_csv(url, dtype={
    "id": "string",
    "title": "string",
    "type": "category",
    "genres": "string",
    "averageRating": "float32",
    "numVotes": "Int32",
    "releaseYear": "Int16"
}, storage_options={"string_storage": "pyarrow"})

print(df.dtypes)
print(f"Uso de memoria: {df.memory_usage(deep=True).sum() / (1024**2):.2f} MB")

id               string[python]
title            string[python]
type                   category
genres           string[python]
averageRating           float32
numVotes                  Int32
releaseYear               Int16
dtype: object
Uso de memoria: 219.42 MB
